File: app/src/main/java/com/example/healthtracker/feature/main/view/MainAdapter.java
package com.example.healthtracker.feature.main.view;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.healthtracker.R;

import java.util.List;

public class MainAdapter extends RecyclerView.Adapter<MainAdapter.MainViewHolder> {

    private List<MainItem> items;

    public MainAdapter(List<MainItem> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public MainViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.main_item, parent, false);
        return new MainViewHolder(view);
    }

    @Override
    public void onBindViewHolder(MainViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class MainViewHolder extends RecyclerView.ViewHolder {
        public MainViewHolder(View itemView) {
            super(itemView);
        }

        public void bind(MainItem item) {
            TextView itemText = itemView.findViewById(R.id.item_text);
            ImageView itemIcon = itemView.findViewById(R.id.item_icon);

            itemText.setText(item.getText());
            itemIcon.setImageResource(item.getIcon());

            itemView.setOnClickListener(v -> item.onClickListener());
        }
    }
}

File: app/src/main/java/com/example/healthtracker/feature/main/view/MainAdapter.kt
package com.example.healthtracker.feature.main.view

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.healthtracker.R

class MainAdapter(
    private var items: List<MainItem>,
    private val callback: (id: Int) -> Unit
): RecyclerView.Adapter<MainAdapter.MainViewHolder>() {
    override fun onBindViewHolder(holder: MainViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MainViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.main_item, parent, false)
        return MainViewHolder(view)
    }

    inner class MainViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(item: MainItem) {
            val itemText: TextView = itemView.findViewById(R.id.item_text)
            val itemIcon: ImageView = itemView.findViewById(R.id.item_icon)
            itemText.setText(item.text)
            itemIcon.setImageResource(item.icon)
            itemView.setOnClickListener {
                callback(item.id)
            }
        }
    }
}